### PR TITLE
OR Query public API

### PR DESCRIPTION
**Pull request by MarkDuckworth**
Adding the OR Query public API

### CODE REVIEW

**Comment by MarkDuckworth** [comment]
@egilmorez please take a look at these documentation changes for the public API for the upcoming OR query feature.

**Review by egilmorez — COMMENTED** [review]
One high-level question before we pass the review to Mark A., thanks!

File: packages/firestore/test/integration/api/query.test.ts
Comment (ehsannas) [inline packages/firestore/test/integration/api/query.test.ts:1838]: why has `c` been added to the test docs?

File: packages/firestore/test/integration/api/query.test.ts
Comment (ehsannas) [inline packages/firestore/test/integration/api/query.test.ts:1535]: the name of the test can be updated now:
`can use or queries with not-in`

File: docs-devsite/firestore_lite.querycompositefilterconstraint.md
Comment (egilmorez) [inline docs-devsite/firestore_lite.querycompositefilterconstraint.md:1]: Did I forget to press a button earlier today . . . ??

Anyway, my question was: why are these output files included in the PR? Generally we do those in a follow-on CL.  Thanks!

File: packages/firestore/test/integration/api/query.test.ts
Comment (MarkDuckworth) [reply packages/firestore/test/integration/api/query.test.ts:1461]: fixed

File: packages/firestore/test/integration/api/query.test.ts
Comment (MarkDuckworth) [reply packages/firestore/test/integration/api/query.test.ts:1536]: Thanks. Removed

File: packages/firestore/test/integration/api/query.test.ts
Comment (MarkDuckworth) [reply packages/firestore/test/integration/api/query.test.ts:1550]: Removed

File: packages/firestore/test/integration/api/query.test.ts
Comment (MarkDuckworth) [reply packages/firestore/test/integration/api/query.test.ts:1559]: Removed

File: packages/firestore/test/integration/api/query.test.ts
Comment (MarkDuckworth) [reply packages/firestore/test/integration/api/query.test.ts:1581]: Removed

File: packages/firestore/test/integration/api/query.test.ts
Comment (MarkDuckworth) [reply packages/firestore/test/integration/api/query.test.ts:1596]: Removed

File: packages/firestore/test/integration/api/query.test.ts
Comment (MarkDuckworth) [reply packages/firestore/test/integration/api/query.test.ts:1535]: Updated

File: packages/firestore/test/integration/api/query.test.ts
Comment (MarkDuckworth) [reply packages/firestore/test/integration/api/query.test.ts:1838]: It was not added. `it('supports order by equality'` was not modified. The diff is invalid.

File: .changeset/sweet-rats-compete.md
Comment (dwyfrequency) [inline .changeset/sweet-rats-compete.md:2]: Update to 
```
'@firebase/firestore': minor
'firebase': minor
```

File: docs-devsite/firestore_lite.querycompositefilterconstraint.md
Comment (markarndt) [reply docs-devsite/firestore_lite.querycompositefilterconstraint.md:1]: @egilmorez ... wrt to the feature, this content looks fine. Do you want resolution to this discussion before I give the single approval that will unblock?

File: .changeset/sweet-rats-compete.md
Comment (MarkDuckworth) [reply .changeset/sweet-rats-compete.md]: Modified. Thank you for the review.

**Review by ehsannas — APPROVED** [review]

File: docs-devsite/firestore_lite.querycompositefilterconstraint.md
Comment (egilmorez) [reply docs-devsite/firestore_lite.querycompositefilterconstraint.md:1]: Nope, go ahead Mark, thanks!  I'm curious see if a doc CL crosses my "desk."

**Review by egilmorez — APPROVED** [review]
iiuc, all the doc strings are in query.ts and reference.ts, both of which look fine to me.

**Review by dwyfrequency — APPROVED** [review]
LGTM

**Pull request merged**
